subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultFileOperations.java

```diff
@@ -153,7 +155,33 @@ public ConfigurableFileTree fileTree(Map<String, ?> args) {
 
     @Override
     public FileTree zipTree(Object zipPath) {
-        return new FileTreeAdapter(new ZipFileTree(file(zipPath), getExpandDir(), fileSystem, directoryFileTreeFactory, fileHasher), patternSetFactory);
+        return zipTree(zipPath, null);
+    }
+
+    @Override
+    public FileTree zipTree(Map<String, ?> args) {
+        Object zipPath = args.get("file");
+        if (zipPath == null) {
+            throw new InvalidUserDataException("zipTree requires the 'file' argument.");
+        }
+
+        Object metadataCharset = args.containsKey("metadataCharset") ? args.get("metadataCharset") : null;
```

Comment:
```suggestion
        Object metadataCharset = args.get("metadataCharset");
```

Is `containsKey` required?

Reply:
It's not required, thanks!